org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java

```diff
@@ -41,39 +42,22 @@
 /**
  * @author Sven Efftinge - Initial contribution and API
  */
-public class CodeActionService implements ICodeActionService2 {
+public class CodeActionService extends QuickFixCodeActionService {
 	@Inject
 	private IChangeSerializer serializer;
 
 	@Override
 	public List<Either<Command, CodeAction>> getCodeActions(ICodeActionService2.Options options) {
-		List<Either<Command, CodeAction>> actions = new ArrayList<>();
+		List<Either<Command, CodeAction>> actions = super.getCodeActions(options);
 		for (Diagnostic d : options.getCodeActionParams().getContext().getDiagnostics()) {
 			Object code = d.getCode().get();
-			if (TestLanguageValidator.INVALID_NAME.equals(code)) {
-				actions.add(Either.forLeft(fixInvalidName(d, options)));
-			} else if (TestLanguageValidator.UNSORTED_MEMBERS.equals(code)) {
+			if (TestLanguageValidator.UNSORTED_MEMBERS.equals(code)) {
 				actions.add(Either.forRight(fixUnsortedMembers(d, options)));
 			}
 		}
 		return actions;
 	}
 
-	private Command fixInvalidName(Diagnostic d, ICodeActionService2.Options options) {
```

Comment:
please restore. otherwise we would have code action as command untested.

Reply:
Ok, then I'll do as you suggested

Reply:
i assume the test (framework) can handle multiple results. correct?

Reply:
yes